src/FondOfSpryker/Glue/ProductListsRestApi/Processor/ProductLists/ProductListsReader.php

```diff
@@ -0,0 +1,244 @@
+                $productListResponseTransfer->getProductList(),
+                new RestProductListsAttributesTransfer()
+            );
+
+        return $this->addRestProductListsAttributesTransferToResponse($restProductListsAttributesTransfer, $restResponse);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListTransfer $productListTransfer
+     * @param \Generated\Shared\Transfer\RestUserTransfer|null $restUserTransfer
+     *
+     * @return bool
+     */
+    protected function productListIsFromCustomerOrCompanyUser(ProductListTransfer $productListTransfer, ?RestUserTransfer $restUserTransfer): bool
+    {
+        if ($restUserTransfer === null) {
+            return false;
+        }
+
+        if ($this->productListCustomerRelationHasRestUser($productListTransfer->getProductListCustomerRelation(), $restUserTransfer)) {
+            return true;
+        }
+
+        if ($this->productListCompanyUserRelationshipHasRestUserCompanyId($productListTransfer->getProductListCompanyRelation(), $restUserTransfer)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListCustomerRelationTransfer|null $productListCustomerRelationTransfer
+     * @param \Generated\Shared\Transfer\RestUserTransfer $restUserTransfer
+     *
+     * @return bool
+     */
+    protected function productListCustomerRelationHasRestUser(?ProductListCustomerRelationTransfer $productListCustomerRelationTransfer, RestUserTransfer $restUserTransfer): bool
+    {
+        if ($productListCustomerRelationTransfer === null) {
+            return false;
+        }
+
+        return in_array($restUserTransfer->getSurrogateIdentifier(), $productListCustomerRelationTransfer->getCustomerIds(), true);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListCompanyRelationTransfer|null $productListCompanyRelationTransfer
+     * @param \Generated\Shared\Transfer\RestUserTransfer $restUserTransfer
+     *
+     * @return bool
+     */
+    protected function productListCompanyUserRelationshipHasRestUserCompanyId(?ProductListCompanyRelationTransfer $productListCompanyRelationTransfer, RestUserTransfer $restUserTransfer): bool
+    {
+        if ($productListCompanyRelationTransfer === null) {
+            return false;
+        }
+
+        $customerTransfer = (new CustomerTransfer())
+            ->setCustomerReference($restUserTransfer->getNaturalIdentifier());
+
+        $companyUserCollection = $this->companyUserClient->getActiveCompanyUsersByCustomerReference($customerTransfer);
+
+        foreach ($companyUserCollection->getCompanyUsers() as $companyUserTransfer) {
+            if ($companyUserTransfer->getCompany() === null) {
+                continue;
+            }
+
+            if (in_array($companyUserTransfer->getCompany()->getIdCompany(), $productListCompanyRelationTransfer->getCompanyIds(), true)) {
```

Comment:
you should split the parameters in more than on line it exceds the maxmium length limit

 if (in_array($companyUserTransfer->getCompany()->getIdCompany(), $productListCompanyRelationTransfer->getCompanyIds(), true)) {